Sources/Commands/Workspace.swift

```diff
@@ -216,6 +216,30 @@ public class Workspace {
         }
     }
 
+    /// Cleans the build artefacts from workspace data.
+    func clean() throws {
+        // These are the things we don't want to remove while cleaning.
+        let protectedAssets: Set<AbsolutePath> = [
```

Comment:
Should we change this to compute a list of names, and assert that all the inputs are direct subpaths of `dataPath`? If we change where they go to be subdirectories of `dataPath` this code will silently remove them (since the name won't match).